sdk/python/touca/cli/server.py

```diff
@@ -45,98 +71,71 @@ def find_install_dir():
         if find_compose_file(default_dir)
         else Path(ask("Where is Touca installed?", default_dir)).expanduser().absolute()
     )
-    compose_file = find_compose_file(install_dir)
-    if not compose_file:
+    if not find_compose_file(install_dir):
         raise RuntimeError(f"Touca server is not installed in {install_dir}")
     return install_dir
 
 
-def uninstall_instance(install_dir: Path):
+def uninstall_instance(compose: Compose, install_dir: Path):
     from shutil import rmtree
 
     compose_file = find_compose_file(install_dir)
     if not compose_file:
         return
     logger.info("Stopping running containers")
-    if run_compose(compose_file, "stop"):
+    if compose.run(compose_file, "stop"):
         raise RuntimeError("failed to stop running containers")
     logger.info("Removing stopped containers")
-    if run_compose(compose_file, "down"):
+    if compose.run(compose_file, "down"):
         raise RuntimeError("failed to remove stopped containers")
     for x in ["data", "logs", "ops"]:
         rmtree(install_dir.joinpath(x), ignore_errors=True)
 
 
-def upgrade_instance(install_dir: Path):
+def upgrade_instance(compose: Compose, install_dir: Path):
     compose_file = find_compose_file(install_dir)
     logger.info("Updating docker images")
-    if run_compose(compose_file, "pull"):
+    if compose.run(compose_file, "pull"):
         raise RuntimeError("failed to update docker images")
     logger.info("Starting new containers")
-    if run_compose(compose_file, "up", "-d", "--remove-orphans"):
+    if compose.run(compose_file, "up", "-d", "--remove-orphans"):
         raise RuntimeError("failed to start new containers")
 
 
 def ask(question: str, default: str = None):
     from colorama import Style
 
     msg = f' {Style.DIM}(default is "{default}"){Style.RESET_ALL}' if default else ""
-    output = input(f"$ {question}{msg}\n> ")
+    try:
+        output = input(f"$ {question}{msg}\n> ")
+    except KeyboardInterrupt:
+        raise RuntimeError()
```

Comment:
Unrelated change: when we prompt user and they ctrl-c for any reason, this change helps us gracefully handle the raised exception.